File: src/main/java/net/felsstudio/fels/exceptions/LexerException.java
package main.java.net.felsstudio.fels.exceptions;

import main.java.net.felsstudio.fels.parser.Pos;

/**
 *
 * @author felek
 */
public final class LexerException extends RuntimeException {

    public LexerException(String message) {
        super(message);
    }

    public LexerException(Pos pos, String message) {
        super(pos.format() + " " + message);
    }
}

File: src/net/felsstudio/fels/exceptions/LexerException.kt
package net.felsstudio.fels.exceptions

/**
 *
 * @author felek
 */
class LexerException : RuntimeException {
    constructor(message: String?) : super(message)

    constructor(row: Int, col: Int, message: String) : super("[$row:$col] $message")
}